server/src/main/java/io/druid/client/indexing/IndexingServiceClient.java

```diff
@@ -79,6 +88,45 @@ public void mergeSegments(List<DataSegment> segments)
     runQuery(new ClientAppendQuery(dataSource, segments));
   }
 
+  public String hadoopMergeSegments(
+      String dataSource,
+      List<Interval> intervalsToReindex,
+      AggregatorFactory[] aggregators,
+      DruidCoordinatorHadoopMergeConfig hadoopMergeConfig
+  )
+  {
+    final InputStream queryResponse = runQuery(
+        new ClientHadoopIndexQuery(
+            String.format(
```

Comment:
needs uuid added if it is going to be unique, but to fit in with the other maintenance-like tasks, it should at least use `AbstractTask.makeId`